src/frame/sdk/UserApi.ts

```diff
@@ -25,6 +25,28 @@ export function deauthorizeDevice() {
     );
 }
 
+/**
+ * Rotate users current private key by taking their current passcode and using it to derive a key to decrypt their user private key.
+ * Then generates and augmentation factor and subtracts that augmentation factor from the users private key. The new private key is then
+ * encrypted. The new encrypted private key and the augmetnation factor is then passed as an authorized api request. The server then
+ * validates the request and augments the server side private key.
+ * @param {string} passcode Users current passcode
+ */
+export function rotateUserMasterKey(passcode: string) {
+    const encryptedPrivateUserKey = ApiState.encryptedUserKey();
+    const payload: WMT.RotateUserPrivateKeyWorkerRequest = {
+        type: "ROTATE_USER_PRIVATE_KEY",
+        message: {
+            passcode,
+            encryptedPrivateUserKey,
+            keySalt: sliceArrayBuffer(encryptedPrivateUserKey, 0, CryptoConstants.SALT_LENGTH),
```

Comment:
I'm not sure there's any reason to slice off the keySalt here since what we're slicing here is already full sent in the `encryptedPrivateUserKey`. The code within the worker should just do the work to slice off the salt.

Reply:
done